### PR TITLE
Vote 801 -> cypress installation 

### PR DESCRIPTION
**Pull request by clmedders**
link to ticket -> https://bixal-projects.atlassian.net/browse/VOTE-801

### Purpose of PR: 
This pr will add cypress to vote.gov drupal repo 

### Extra Info on work:

- I created a `testing` folder... in order to move the `cypress.config.js`, `package.json` and `package-lock.json` into the cypress folder it needed an extra one on top of it in order for the tests to work.  I was not able to just add them to the cypress folder and everything still works 
- there is a sample spec file to show that everything is working.. this just simply loads the page and checks that it has loaded, once there is more migrated in this will be removed.
- there will be future configuration needed such as scripts and other tests added but will do that on an as-needed basis. 

### How to test:

1. pull down branch 
2. start local env 
3. cd into `testing` folder 
4. `npm run cy:test` this will run test headless 
5. `npm run cy:open` this will open the cypress runner 

for the last 2 steps ensure that the test is passing as expected

### CODE REVIEW

File: testing/cypress.config.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    },
+    chromeWebSecurity: false,
+    responsetimeout: 10000,
+    "blockHosts": ["www.google-analytics.com", "ssl.google-analytics.com"],
+    
+    setupNodeEvents(on, config) {
+      // implement node event listeners here
+    },
+  },
+});
```
Comment (clmedders) [inline testing/cypress.config.js:21]: most of this was brought over from the old vote.gov set up

File: testing/package.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "vote-gov-drupal",
```
Comment (rayestrada) [inline testing/package.json:2]: update name to `"Vote.gov Cypress testing suite"`

File: testing/package.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  "version": "1.0.0",
+  "description": "## Requirements",
+  "main": "index.js",
+  "dependencies": {},
+  "devDependencies": {
+    "cypress": "^12.3.0"
+  },
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
```
Comment (rayestrada) [inline testing/package.json:11]: Can we make `test` script run `cy:test`?

Comment (rayestrada) [reply testing/package.json:11]: You might have to reorder the list of scripts.

**Review by rayestrada — CHANGES_REQUESTED** [review]
Christian can you add a `.nvmrc` file in here to define which version of node. We are using node 18 for the theme. I would suggest using the same version.

File: testing/.nvmrc [hunk 4]
```diff
@@ -0,0 +1 @@
+18
```
Comment (clmedders) [inline testing/.nvmrc:1]: added file!

File: testing/package.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "vote.gov-cypress-testing-suite",
```
Comment (clmedders) [inline testing/package.json:2]: went to update the name with what you suggested but it yelled at me for format and wanted it in this format

File: testing/package.json [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  "main": "index.js",
+  "dependencies": {},
+  "devDependencies": {
+    "axe-core": "^4.6.3",
+    "cypress": "^12.3.0",
+    "cypress-axe": "^1.3.0"
+  },
+  "scripts": {
+    "cy:open": "cypress open",
+    "cy:test": "cypress run --spec 'cypress/e2e/tests/**/*'",
```
Comment (clmedders) [inline testing/package.json:14]: so i had a `cy.test` already but i think that one just came with the file when I ran the `npm init` command.  i took out the other `test` in the file but here is the `cy:test` that will run the cypress test headless.

**Review by rayestrada — APPROVED** [review]

**Pull request merged**